test/IntegrationTests/Helpers/MockProfilesCollector.cs

```diff
@@ -0,0 +1,193 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#if NET6_0_OR_GREATER
+
+using System.Collections.Concurrent;
+using System.Text;
+using Microsoft.AspNetCore.Http;
+using OpenTelemetry.Proto.Collector.Profiles.V1;
+using OpenTelemetry.Proto.Profiles.V1;
+using Xunit.Abstractions;
+
+namespace IntegrationTests.Helpers;
+
+public class MockProfilesCollector : IDisposable
+{
+    private readonly ITestOutputHelper _output;
+    private readonly TestHttpServer _listener;
+
+    private readonly List<Expectation> _expectations = new();
+    private readonly BlockingCollection<Collected> _profilesSnapshots = new(10); // bounded to avoid memory leak; contains protobuf type
+
+    public MockProfilesCollector(ITestOutputHelper output, string host = "localhost")
```

Comment:
`host` parameter can be removed

Reply:
9c2023a82edf585337ff7796050e964936eeabed